src/components/checkbox/checkbox.js

```diff
@@ -1,29 +1,144 @@
+  /**
+   * Checkbox UI.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as a copy button UI.
+   */
+  
+  constructor(element, options) {
+    super(element, options);
+    this.element.addEventListener('click', () => this.handleClick());
+    this.element.addEventListener('focus', () => this.handleFocus());
+    this.element.addEventListener('blur', () => this.handleBlur());
+    
+    this.indeterminateCheckbox();
+    this.initCheckbox();
+  }
+  
+  handleClick() {
+    
+    if (this.element.checked) {
+      this.element.setAttribute('checked', '');
+      this.element.setAttribute('aria-checked', 'true');
+
+      //Adds class for nested checkboxes inside labels
+      if (this.element.parentElement.classList.contains('bx--checkbox-label')) {
+        this.element.parentElement.classList.add('bx--checkbox-label__checked');
+        this.element.parentElement.classList.remove('bx--checkbox-label__indeterminate');
+      }
     } else {
-      element.removeAttribute('checked');
+      this.element.removeAttribute('checked');
+      this.element.setAttribute('aria-checked', 'false');
+      
+      //Removes class for nested checkboxes inside labels
+      if (this.element.parentElement.classList.contains('bx--checkbox-label')) {
+        this.element.parentElement.classList.remove('bx--checkbox-label__checked', 'bx--checkbox-label__indeterminate');
+      }
     }
   }
-}
 
-/**
- * Watches for change in checkbox in the given document and force changing `checked` attribute
- * so that DOM mutation observer in {@link https://www.npmjs.com/package/svgxuse svgxuse} is triggered.
- * @param {Document} [doc=document] The document object to watch for.
- * @returns {Handle} The handle to release the event listener.
- */
-function initCheckbox(doc = document) {
-  doc.removeEventListener('change', listener); // In case this function has been called earlier
-  doc.addEventListener('change', listener);
-  return {
-    release() {
-      doc.removeEventListener('change', listener);
-      return null;
-    },
+  handleFocus() {
+    if (this.element.parentElement.classList.contains('bx--checkbox-label')) {
+      this.element.parentElement.classList.add('bx--checkbox-label__focus');
+    }
+  }
+
+  handleBlur() {
+    if (this.element.parentElement.classList.contains('bx--checkbox-label')) {
+      this.element.parentElement.classList.remove('bx--checkbox-label__focus');
+    }
+  }
+
+  /**
+   * Sets the new checkbox state.
+   * @param {boolean|string} [state]
+   *   The new checkbox state to set. `mixed` to put checkbox in indeterminate state.
+   *   If omitted, this method simply makes the style reflect `aria-checked` attribute.
+   */
+  setState(state) {
+    if (state === undefined || stateChangeTypes[state] === undefined) {
+      console.log('Give us a state of checked, unchecked or mixed');
```

Comment:
Great that you put argument check here! For invalid argument that should stop operation, we typically throw (e.g. https://github.com/carbon-design-system/carbon-components/blob/f748b2c/src/components/loading/loading.js#L29) and great to have the code here align to that convention.